autogen/agentchat/conversable_agent.py

```diff
@@ -601,12 +603,16 @@ def generate_oai_reply(
             return False, None
         if messages is None:
             messages = self._oai_messages[sender]
-
-        # TODO: #1143 handle token limit exceeded error
-        response = oai.ChatCompletion.create(
-            context=messages[-1].pop("context", None), messages=self._oai_system_message + messages, **llm_config
-        )
-        return True, oai.ChatCompletion.extract_text_or_function_call(response)[0]
+        while True:
+            try:
+                response = oai.ChatCompletion.create(
+                    context=messages[-1].pop("context", None), messages=self._oai_system_message + messages, **llm_config
+                )
+                return True, oai.ChatCompletion.extract_text_or_function_call(response)[0]
+            except openai.error.InvalidRequestError as e:
+                if len(messages) < 5:
+                    raise e
+                del messages[2], messages[3]
```

Comment:
The changes look fine but will be good to have a small comment to make the intent clear. This is done to overcome the token limit in case where len(messages) > 5 yet there is a **InvalidRequestError**

Reply:
Agreed. Added comments about why and how.